src/main/kotlin/com/github/vol0n/utbotcppclion/ui/UTBotProgress.kt

```diff
@@ -0,0 +1,47 @@
+package com.github.vol0n.utbotcppclion.ui
+
+import com.intellij.openapi.progress.TaskInfo
+import com.intellij.openapi.progress.util.AbstractProgressIndicatorExBase
+import com.intellij.openapi.wm.ex.StatusBarEx
+import com.intellij.openapi.wm.ex.WindowManagerEx
+import kotlinx.coroutines.Job
+
+class UTBotRequestProgressIndicator(val name: String, var requestJob: Job? = null) : AbstractProgressIndicatorExBase() {
+    val task = UTBotRequestTaskInfo(name)
+
+    init {
+        isIndeterminate = false
+    }
+
+    override fun start() {
+        val frame = WindowManagerEx.getInstanceEx().findFrameFor(null) ?: return
+        val statusBar = frame.statusBar as? StatusBarEx ?: return
+        statusBar.addProgress(this, task)
+        super.start()
+    }
+
+    override fun stop() {
+        requestJob?.cancel()
+        super.stop()
+    }
+
+    fun complete() {
```

Comment:
Maybe just use finish instead of adding `complete()`?

Reply:
using complete you don't need to pass any parameters